packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -48,18 +48,124 @@ export class ArtifactCreator {
         renderer => ({...renderer, file: `${renderer.name}-golden.png`}));
   }
 
-  async captureAndAnalyzeScreenshots(
-      scenarioWhitelist: Set<string>|null = null) {
+  async compareRenderers(scenario: ScenarioConfig) {
+    const analysisResults: AnalysisResults = [];
+    const {rootDirectory, outputDirectory, goldens} = this;
+    const {name: scenarioName, exclude, dimensions} = scenario;
+
+    console.log(
+        `Start to compare model-viewer's golden with other renderers' goldens:`);
+
+    const modelViewerIndex = 0;
+    const modelViewerGoldenPath = join(
+        rootDirectory, 'goldens', scenarioName, goldens[modelViewerIndex].file);
+    let modelViewerGolden;
+    try {
+      modelViewerGolden = await fs.readFile(modelViewerGoldenPath);
+    } catch (error) {
+      throw new Error(`❌ Failed to read model-viewer's ${
+          scenarioName} golden! Error message: ${error.message}`);
+    }
+
+    await fs.writeFile(
+        join(outputDirectory, scenarioName, goldens[modelViewerIndex].file),
+        modelViewerGolden);
```

Comment:
What file are we writing here? What's it for?

Reply:
These golden images are saved to the result folder which are use to show in the result-viewer page. I guess the reason is sometimes you only run fidelity test on specified scenarios, so saving to another folder instead of just reading it from the golden folder can help distinguish which ones to show in the local result-viewer page. (probably there are some other reasons i did get). I have just added a comment here, but do you want me to refactor it so that the result page find the golden images from the golden folder directly?

Reply:
Thank you, I'd completely forgotten about that. No, let's leave it as is; thanks for the comment.